runtime/src/cryptographic_mailbox.rs

```diff
@@ -501,4 +502,37 @@ impl Commands {
             data,
         }))
     }
+
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    #[inline(never)]
+    pub(crate) fn random_stir(
+        drivers: &mut Drivers,
+        cmd_bytes: &[u8],
+    ) -> CaliptraResult<MailboxResp> {
+        if cmd_bytes.len() > core::mem::size_of::<CmRandomStirReq>() {
+            Err(CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS)?;
+        }
+        let mut cmd = CmRandomStirReq::default();
+        cmd.as_mut_bytes()[..cmd_bytes.len()].copy_from_slice(cmd_bytes);
+
+        let size = (cmd.input_size as usize)
+            .next_multiple_of(MAX_SEED_WORDS * 4)
+            .min(MAX_CMB_DATA_SIZE);
+        if size > MAX_CMB_DATA_SIZE {
+            Err(CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS)?;
+        }
+
+        let mut additional_data = [0u32; MAX_CMB_DATA_SIZE / 4];
+        for (i, chunk) in cmd.input[..size].chunks_exact(4).enumerate() {
+            // avoid panic even though this is impossible
+            if i >= additional_data.len() {
+                break;
+            }
+            additional_data[i] = u32::from_be_bytes(chunk.try_into().unwrap());
```

Comment:
Why swap endianness? We're stirring the same data either way, right?

Reply:
In a cryptographic sense it doesn't matter at all.

But it's possible NIST might care that the bits that are brought in are loaded into the block cipher in the correct orientation.

Since we have to convert between `u8` and `u32` anyway, I thought we might as well swap endianess. :shrug: